odbc/src/api/functions.rs

```diff
@@ -1785,11 +1785,441 @@ pub unsafe extern "C" fn SQLGetInfo(
+                SQL_INFO_Y,
+                info_value_ptr as *mut WChar,
+                buffer_length as usize,
+                string_length_ptr,
+            );
+        }
+        // SQL_OWNER_USAGE
+        InfoType::OwnerUsage => {
+            // SQL_OWNER_USAGE is replaced by SQL_SCHEMA_USAGE in newer
+            // ODBC versions. They use the same numeric value.
+            //
+            // MongoSQL does not have a concept of a "schema" which is a
+            // level above "catalog" (or "database") in the data hierarchy.
+            return i16_len::set_output_fixed_data(&SQL_U32_ZERO, info_value_ptr, string_length_ptr);
+        }
+        // SQL_CATALOG_USAGE
+        InfoType::CatalogUsage => {
+            // TODO: (code review) there is no way to indicate this is read-only. The entire DML bitmask
+            //       must be used to indicate we support SELECT. However, that also includes INSERT,
+            //       UPDATE, and DELETE.
+            return i16_len::set_output_fixed_data(&SQL_CU_DML_STATEMENTS, info_value_ptr, string_length_ptr);
+        }
+        // SQL_SPECIAL_CHARACTERS
+        InfoType::SpecialCharacters => {
+            // TODO: (code review) The docs say:
+            //  - "A character string that contains all special characters (that is,
+            //     all characters except a through z, A through Z, 0 through 9, and
+            //     underscore) that can be used in an identifier name, such as a table
+            //     name, column name, or index name, on the data source. For example,
+            //     "#$^". If an identifier contains one or more of these characters, the
+            //     identifier must be a delimited identifier."
+            //  - MongoSQL's grammar defines an identifier character as [^\x00], as in, literally any non-null character.
+            //    Surely we aren't meant to return a "special characters" string that contains every character, are we?
+        }
+        // SQL_MAX_COLUMNS_IN_GROUP_BY
+        InfoType::MaxColumnsInGroupBy => {
+            // MongoSQL does not have an explicit maximum number of
+            // columns allowed in a GROUP BY clause.
+            return i16_len::set_output_fixed_data(&SQL_U16_ZERO, info_value_ptr, string_length_ptr);
+        }
+        // SQL_MAX_COLUMNS_IN_ORDER_BY
+        InfoType::MaxColumnsInOrderBy => {
+            // MongoSQL does not have an explicit maximum number of
+            // columns allowed in a ORDER BY clause.
+            return i16_len::set_output_fixed_data(&SQL_U16_ZERO, info_value_ptr, string_length_ptr);
+        }
+        // SQL_MAX_COLUMNS_IN_SELECT
+        InfoType::MaxColumnsInSelect => {
+            // MongoSQL does not have an explicit maximum number of
+            // columns allowed in a SELECT clause.
+            return i16_len::set_output_fixed_data(&SQL_U16_ZERO, info_value_ptr, string_length_ptr);
+        }
+        // SQL_TIMEDATE_ADD_INTERVALS
```

Comment:
Note: Until we support TIMESTAMPADD and TIMESTAMPDIFF this will not be used, but it's good to have it ready.

Reply:
Left this implemented but updated the comment to reflect it will not be used. Should I actually remove the return value and instead return 0?